configure.ac

```diff
@@ -1654,6 +1654,49 @@ if test x$platform_android = xyes; then
 fi
 
 if test x$host_win32 = xno; then
+	dnl *************************************
+	dnl *** Checks for time capabilities ***
+	dnl *************************************
+
+	AC_MSG_CHECKING(for CLOCK_MONOTONIC)
+	AC_TRY_RUN([
+		#include <stdlib.h>
+		#include <time.h>
+		#include <sys/time.h>
+
+		int main ()
+		{
+			struct timespec ts;
+			return clock_gettime(CLOCK_MONOTONIC, &ts);
+		}
+	],[
+		AC_MSG_RESULT(yes)
+		AC_DEFINE(HAVE_CLOCK_MONOTONIC, 1, [CLOCK_MONOTONIC])
+	], [
+		AC_MSG_RESULT(no)
+	])
+
+	AC_MSG_CHECKING(for CLOCK_MONOTONIC_COARSE)
+	AC_TRY_RUN([
+		#include <stdlib.h>
+		#include <time.h>
```

Comment:
Should probably instead just compile (not run) extern const int foo; extern const int foo = clock_monotic; You don't need main and you don't need to link or run. This is more cross build friendly and more efficient. Autoconf is slow.

Reply:
Indeed, we should not be adding new `AC_TRY_RUN` calls to configure unless the check absolutely can't function without being run. And even in that case, it should probably be a run-time decision in the C code (as we do for e.g. some WSL workarounds). `AC_TRY_RUN` is almost always the wrong thing to use.

Reply:
AC_TRY_RUN doesn't work with cross compiler and we'd have to special case all the new checks there as well.